eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -247,49 +247,50 @@ jobs:
 
-#  - job: 'SmokeTest'
-#    strategy:
-#      matrix:
-#        # Build SDK libraries as frameworks, dynamically link to copies stored within the app bundle
-#        frameworks:
-#          PodsLinkage: frameworks
-#
-#        # Build SDK libraries as static libs (.a), statically link them into the app binary
-#        # This is currently broken, probably because Calling is pre-built as a framework
-#        # static:
-#        #   PodsLinkage: static
-#
-#        # Build SDK libraries as frameworks, but then statically link them into the app binary
-#        # Calling is still dynamically linked, probably because Calling is pre-built as a framework
-#        staticframeworks:
-#          PodsLinkage: staticframeworks
-#
-#    variables:
-#      - template: ../variables/globals.yml
-#
-#    pool:
-#      vmImage: '$(OSVmImage)'
-#
-#    steps:
-#      - script: sudo xcode-select --switch /Applications/Xcode_$(XcodeVersion).app
-#        displayName: 'Use Xcode $(XcodeVersion)'
-#
-#      - script: |
-#          cd examples/AzureSDKSmokeTest
-#          pod install
-#        env:
-#          PODS_LINKAGE: $(PodsLinkage)
-#        displayName: 'Install dependencies'
-#
-#      - script: |
-#          set -o pipefail
-#          cd examples/AzureSDKSmokeTest
-#          xcodebuild build \
-#                     -workspace AzureSDKSmokeTest.xcworkspace \
-#                     -scheme AzureSDKSmokeTest \
-#                     -sdk iphonesimulator \
-#                     -configuration Debug | xcpretty -c
-#        displayName: 'Build smoke test [iphonesimulator]'
+  - job: 'SmokeTest'
+    strategy:
+      matrix:
+        # Build SDK libraries as frameworks, dynamically link to copies stored within the app bundle
+        frameworks:
+          PodsLinkage: frameworks
+
+        # Build SDK libraries as static libs (.a), statically link them into the app binary
+        # This is currently broken, probably because Calling is pre-built as a framework
+        # static:
+        #   PodsLinkage: static
+
+        # Build SDK libraries as frameworks, but then statically link them into the app binary
+        # Calling is still dynamically linked, probably because Calling is pre-built as a framework
+        staticframeworks:
+          PodsLinkage: staticframeworks
+
+    variables:
+      - template: ../variables/globals.yml
+
+    pool:
+      vmImage: '$(OSVmImage)'
+
+    steps:
+      - script: sudo xcode-select --switch /Applications/Xcode_$(XcodeVersion).app
+        displayName: 'Use Xcode $(XcodeVersion)'
+
+      - script: |
+          cd examples/AzureSDKSmokeTest
+          export PODFILE_GIT_REPO=$(curl -s https://api.github.com/repos/Azure/azure-sdk-for-ios/pulls/$(system.pullRequest.pullRequestNumber) | jq -r '.head.repo.full_name')
+          export PODFILE_GIT_BRANCH=$(curl -s https://api.github.com/repos/Azure/azure-sdk-for-ios/pulls/$(system.pullRequest.pullRequestNumber) | jq -r '.head.ref')
```

Comment:
I don't love the fact that this calls a public API twice when I feel once would suffice, but bash is such a nightmare that I'm leaving it unless @mitchdenny or @bsiegel can suggest better syntax.

Also, any suggestions for how to avoid calling the Github API would be very welcome.

Reply:
The only thing I was thinking was that if one of the pre-defined variables in the pipeline already had this detail, but I think that they'll return the details for Azure/azure-sdk-for-ios rather than the fork that you are submitting the PR from.

Reply:
I was able to omit calling the github API by detecting the CI condition and have the Podfile point to the local podspec files that are cloned from the branch/PR.